### PR TITLE
[FLINK-11679][table] Create Blink SQL planner and runtime modules

### PR DESCRIPTION
**Pull request by twalthr**
## What is the purpose of the change

This commit adds flink-table-planner-blink and flink-table-runtime-blink
modules. Those modules are work-in-progress and are not connected to the
API yet. This happens at a later stage (see FLINK-11452). We aim to make
the runtime module Scala-free.

Initial dependencies have been added. Optimization with Calcite, code
generation, and translation to operators are performed in the planner
module.

Actual execution classes and code compilation are performed in the runtime
module.

## Brief change log

- New modules added

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not documented

### CODE REVIEW

**Comment by KurtYoung** [comment]
@flinkbot approve all

File: flink-table/flink-table-planner-blink/pom.xml [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+	<packaging>jar</packaging>
+
+	<dependencyManagement>
+		<dependencies>
+			<!-- Common dependency of calcite-core and flink-* -->
+			<dependency>
+				<groupId>com.google.guava</groupId>
+				<artifactId>guava</artifactId>
+				<version>19.0</version>
```
Comment (KurtYoung) [inline flink-table/flink-table-planner-blink/pom.xml]: Does it make sense to use property like guava.version, to keep all version synced.

Comment (KurtYoung) [reply flink-table/flink-table-planner-blink/pom.xml]: Oh, i just noticed guava version has been defined in root pom, and the version is 18.0. Do we need 19.0 for table?

File: flink-table/flink-table-planner-blink/pom.xml [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+				<exclusion>
+					<groupId>commons-dbcp</groupId>
+					<artifactId>commons-dbcp</artifactId>
+				</exclusion>
+				<exclusion>
+					<groupId>com.esri.geometry</groupId>
+					<artifactId>esri-geometry-api</artifactId>
+				</exclusion>
+			</exclusions>
+		</dependency>
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/pom.xml]: We will add test dependencies when needed?

File: flink-table/flink-table-planner-blink/pom.xml (hunk 1 above)
Comment (twalthr) [reply flink-table/flink-table-planner-blink/pom.xml]: Calcite does only support a certain range of Guava versions. I think it makes sense to hard code the version here next to Calcite. About Janino we just had a discussion in #7407 but I like the idea of defining the Janino version globally in a variable. Will do that.

File: flink-table/flink-table-planner-blink/pom.xml (hunk 2 above)
Comment (twalthr) [reply flink-table/flink-table-planner-blink/pom.xml]: Yes, there were a lot of test dependencies in `flink-table-planner` and I was unsure what will still be needed. We can readd them when we merge tests.

Comment (wuchong) [reply flink-table/flink-table-planner-blink/pom.xml]: Make sense to me.

**Comment by twalthr** [comment]
@KurtYoung I updated the PR. If there are no objections, I will merge this once the feature branch has been forked off. Contributors can already use the commit to open PRs for FLINK-11488.

**Comment by twalthr** [comment]
The release branch has been forked. I will merge this...

**Pull request closed without merging**